webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -407,6 +417,9 @@ public void push(
     List<SmartlingFile> result =
         Stream.concat(singularFiles, pluralFiles).collect(Collectors.toList());
     resultProcessor.processPush(result, options);
+    meterRegistry
+        .timer("ThirdPartyTMSSmartling.push", "repository", repository.getName())
+        .record(totalStopwatch.stop().elapsed());
```

Comment:
Not sure if you know about this pattern where the "instrumented code" is wrapped in lambda that is passed to `record()`? 

``` meterRegistry
        .timer(
            "PullRunAssetService.saveTextUnitVariantsMultiRow",
            Tags.of("repositoryId", Objects.toString(repository.getId())))
        .record(
            () -> {
              Instant now = Instant.now();
              jdbcTemplate.update(
                  "delete from pull_run_text_unit_variant where pull_run_asset_id = ? and locale_id = ?",
                  pullRunAsset.getId(),
                  localeId);
              Lists.partition(uniqueTmTextUnitVariantIds, BATCH_SIZE)
                  .forEach(
                      tuvIdsBatch ->
                          saveTextUnitVariantsMultiRowBatch(
                              pullRunAsset, localeId, tuvIdsBatch, now));
            });```

Reply:
I'd seen the option but when I tried to use it in some cases I ran into issues around some of the existing implementation variables needing to be `final or effectively final` to be able to be used within a lamda. I then switched to this approach to avoid needing to refactor lots of the existing code to be lamda friendly.

Reply:
With this pattern the behavior will be a bit different if exception are raise. We'd need a try catch to have the same behavior. 

I also see we started to use "retry with lambda" in that class. So i'M wondering if the re-factoring is warranted. Is it hard to do elegantly? 

Let me know your last thoughts, I could merge like that if you feel it is the best approach.

Reply:
I've updated to the lamda metric reporting approach now, all unit tests are passing